Valkyrization

kirkkwang opened the pull request:
Add valkyrie support

alishaevn commented on lib/willow_sword/zip_package.rb:
```diff
@@ -43,7 +43,7 @@ def writeEntries(entries, path, io)
         else
           disk_file = File.open(diskFilePath, "rb")
-          io.get_output_stream(zipFilePath) { |f| f.puts(disk_file.read()) }
+          io.get_output_stream(zipFilePath) { |f| f.write(disk_file.read()) }
```
although [this doc](https://rdoc.info/github/rubyzip/rubyzip/master/Zip/File) shows `f.puts`, willow_sword is currently using version `1.2.1` of rubyzip. according to that version, `f.write` is the [suggested usage](https://github.com/rubyzip/rubyzip/tree/v1.2.1#basic-zip-archive-creation). (same for the `master` branch)

Commit:
🐛 Fix GET requests

This commit will fix the GET requests for collections, works, and
file_sets.  This required to udpate the old active fedora calls to
valkyrie queries.

Commit:
🐛 Fix PUT and PATCH requests

This commit will fix the PUT nad PATCH requests for works and file_sets
with using valkyrie queries.

Commit:
🐛 Fix POST requests

This commit will fix the POST requests for works and file_sets with
using valkyrie queries.

Commit:
Add default work type

This commit will add a default work type to be used if there isn't a
specified one.

Also, searching for the metadata file in either /data or
/data/metadata/.

Commit:
✅ Update dependencies and fix spec

This commit will update the rails dependency to match Hyrax 5's version.
Also adding byebug to the test group to help with debugging.  Lastly,
Fixed a spec that hardcoded a path.
